universal-application-tool-0.0.1/app/services/question/QuestionServiceImpl.java

```diff
@@ -129,6 +131,36 @@ public void discardDraft(Long id) throws InvalidUpdateException {
       throw new InvalidUpdateException("Did not find question in draft version.");
     }
     question.get().save();
+    versionRepositoryProvider.get().updateProgramsForNewDraftQuestion(id);
+  }
+
+  @Override
+  public ImmutableList<QuestionDefinition> getQuestionsForTag(QuestionTag tag) {
+    return questionRepository.getAllQuestionsForTag(tag);
+  }
+
+  @Override
+  public void setExportState(QuestionDefinition questionDefinition, String questionExportState)
+      throws QuestionNotFoundException, InvalidUpdateException {
+    Optional<Question> question =
+        questionRepository.lookupQuestion(questionDefinition.getId()).toCompletableFuture().join();
+    if (question.isEmpty()) {
+      throw new QuestionNotFoundException(questionDefinition.getId());
+    }
+    if (questionExportState.equals(QuestionTag.DEMOGRAPHIC.getValue())) {
+      question.get().removeTag(QuestionTag.DEMOGRAPHIC_PII);
+      question.get().addTag(QuestionTag.DEMOGRAPHIC);
+    } else if (questionExportState.equals(QuestionTag.DEMOGRAPHIC_PII.getValue())) {
+      question.get().removeTag(QuestionTag.DEMOGRAPHIC);
+      question.get().addTag(QuestionTag.DEMOGRAPHIC_PII);
+    } else if (questionExportState.equals("none")) {
+      question.get().removeTag(QuestionTag.DEMOGRAPHIC_PII);
+      question.get().removeTag(QuestionTag.DEMOGRAPHIC);
+    } else {
+      throw new InvalidUpdateException(
+          String.format("Unknown question export state: %s", questionExportState));
+    }
+    question.get().save();
```

Comment:
Suggestion (non-blocking): pull the question into a variable so you don't have to call `get()` for each access

Reply:
Done.